test/e2e/workflow/deployment_test.go

```diff
@@ -89,6 +90,14 @@ var _ = Describe("NetworkAddonsConfig", func() {
 				[]Component{MacvtapComponent},
 			),
 		)
+		It("should deploy prometheus if NetworkAddonsConfigSpec is not empty", func() {
+			isMonitoringDeployed, err := networkaddonsconfig.IsMonitoringAvailable(framework.Global.KubeClient)
+			Expect(err).ToNot(HaveOccurred())
+			if !isMonitoringDeployed {
```

Comment:
why allowing in the tests to be not deployed and skip ?

Reply:
well it depends.. This logic is simply mimicking what we do in the production code.
But what it means in the context of this tests is that if we don't skip this test then k8s providers <1.21 will just fail this test/e2e/workflow lane.
Do you think we should not care of old providers?

Reply:
ye lets not care about them
if and only if we will need it in the future we can reconsider, but i bet we won't need it
so better to be strict + less to maintain

Reply:
I agree. Done